logic/atrium-logic/src/commonMain/kotlin/ch/tutteli/atrium/logic/impl/DefaultIterableLikeAssertions.kt

```diff
@@ -135,9 +138,25 @@ class DefaultIterableLikeAssertions : IterableLikeAssertions {
 
         val assertions = ArrayList<Assertion>(2)
         assertions.add(createExplanatoryAssertionGroup(container, assertionCreatorOrNull))
-        val mismatches = createIndexAssertions(list) { (_, element) ->
-            mismatchIf(allCreatedAssertionsHold(container, element, assertionCreatorOrNull))
+        val mismatches = list.mapIndexedNotNull { index, e ->
+            val assertion = if (assertionCreatorOrNull != null) {
+                container.collectBasedOnSubject(Some(e as E), assertionCreatorOrNull)
+            } else {
+                // TODO: assertionCreatorOrNull should be non-nullable, and this "to be null" should be a part of expectation
+                assertionBuilder.createDescriptive(DescriptionBasic.TO_BE, Text.NULL) {
+                    e == null
+                }
+            }
+
+            assertion.takeIf { mismatchIf(it.holds()) }?.let {
+                assertionBuilder
+                    .feature
+                    .withDescriptionAndRepresentation(TranslatableWithArgs(INDEX, index), e)
+                    .withAssertion(it)
+                    .build()
+            }
```

Comment:
You should modify createIndexAssertions - I suggest you rename it to `createMismatch` as it is used only for this purpose

Reply:
Of course, it should replace the current `createIndexAssertions`, however, I tried to implement something minimal to see what are other changes required.

One of the takeaways, for now, is the current implementation shows both successful and failing assertions, and it does not distinguish between successful and failing ones.
For instance `login: "joe"` was successful, however, it was displayed the same as `password: "qwerty"` which failed.